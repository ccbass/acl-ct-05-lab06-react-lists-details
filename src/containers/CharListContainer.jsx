import React, { useState } from 'react'
import {useAvatarFetchChars} from '../components/characters/CharHooks'
import CharList from '../components/characters/CharList'



function CharListContainer() {
    const {data, loading, changePage} = useAvatarFetchChars()

    return (
        <div>
            <button onClick={() => changePage('DOWN')}>PREV PAGE</button>
            <button onClick={() => changePage('UP')}>NEXT PAGE</button>
            {loading === 'loading' && <h1>L O A D I N G</h1>}
            <CharList characters={data} />
            
        </div>
    )
}



export default CharListContainer

