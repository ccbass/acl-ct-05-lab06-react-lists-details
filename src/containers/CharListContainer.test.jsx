import React from 'react';
import { render, screen,waitFor } from '@testing-library/react';
import CharListContainer from './CharListContainer'
import { BrowserRouter as Router } from 'react-router-dom'
import userEvent from '@testing-library/user-event';


describe('CharListContainer container tests', () => {

    it('runs all container tests', async () => {
      render(<Router><CharListContainer /></Router>);

      // Check for loading text to appear before API resolves
      screen.getByText('L O A D I N G');
  
      // Check that 10 image cards load and display in DOM
      // Compare to snapshots
      const images = await screen.findAllByRole('img')
      expect(images).toHaveLength(10)
      expect(images).toMatchSnapshot()

      // Click next page button
      const nextPage = await screen.getByRole('button', {name: 'NEXT PAGE'})
      userEvent.click(nextPage)

      // Wait for DOM to update and check if new character is listed.
      return waitFor(async () => {
          const character = await screen.findByText('Chow')
          expect(character).toBeInDOM
      })

    });
      
      
  });