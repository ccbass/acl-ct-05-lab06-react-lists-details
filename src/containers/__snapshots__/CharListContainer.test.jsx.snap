// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharListContainer container tests runs all container tests 1`] = `
Array [
  <img
    src="https://vignette.wikia.nocookie.net/avatar/images/c/c3/Analay.png/revision/latest?cb=20140430182201"
  />,
  <img
    src="https://vignette.wikia.nocookie.net/avatar/images/6/6f/Arik.png/revision/latest?cb=20140808184002"
  />,
  <img
    src="https://vignette.wikia.nocookie.net/avatar/images/4/45/Baatar_Jr..png/revision/latest?cb=20150409193510"
  />,
  <img
    src="https://vignette.wikia.nocookie.net/avatar/images/d/d4/Bolin.png/revision/latest?cb=20150406232314"
  />,
  <img />,
  <img
    src="https://vignette.wikia.nocookie.net/avatar/images/8/81/Chit_Sang.png/revision/latest?cb=20121021000707"
  />,
  <img
    src="https://vignette.wikia.nocookie.net/avatar/images/b/b6/Asami_Sato.png/revision/latest?cb=20150407110150"
  />,
  <img
    src="https://vignette.wikia.nocookie.net/avatar/images/b/b6/Bum-Ju.png/revision/latest?cb=20140730103156"
  />,
  <img
    src="https://vignette.wikia.nocookie.net/avatar/images/8/82/Butler.png/revision/latest?cb=20121107115044"
  />,
  <img
    src="https://vignette.wikia.nocookie.net/avatar/images/9/98/Broadsword_man.png/revision/latest?cb=20130630171553"
  />,
]
`;
